pkg/metrics/otel.go

```diff
@@ -46,6 +47,14 @@ type SemConvMetricsRegistry struct {
 
 // NewSemConvMetricRegistry registers all stables semantic conventions metrics.
 func NewSemConvMetricRegistry(ctx context.Context, config *types.OTLP) (*SemConvMetricsRegistry, error) {
+	if os.Getenv("USER") == "" {
+		// If USER environment variable is not set, we are likely running in a container.
+		// Set it to "traefik" so that some libraries (like OpenTelemetry) do not fail to initialize.
+		if err := os.Setenv("USER", "traefik"); err != nil {
+			return nil, fmt.Errorf("could not set USER environment variable: %w", err)
+		}
+	}
+
```

Comment:
Since the same code block is duplicated across three files WDYT about creating a common utility function such as:
```go
func EnsureUserEnvVar() error {
    if os.Getenv("USER") == "" {
        if err := os.Setenv("USER", "traefik"); err != nil {
            return fmt.Errorf("could not set USER environment variable: %w", err)
        }
    }
    return nil
}
```

This would:
- Reduce code duplication
- Make the logic easier to test
- Ensure consistency across all OpenTelemetry initialization points
- Make future changes easier to maintain